### ISSUE TITLE
Document safety issues

### ISSUE THREAD

**Issue by marcandre** [rubocop/rubocop#8431]
It would be nice that Cops that are not safe of have a not safe auto-correction have a section in their doc about this

E.g. for FrozenStringLiteral

```ruby
=== Safety
Unsafe since any code that mutates these literals will fail:
    # bad, not safely corrected:
    str = 'hello'
    str << 'world' # => Fails if magic comment is present.

    # good
    # frozen_string_literal: true

    str = +'hello'  # Notice the '+' to obtain unfrozen litteral here
    puts str << ', world'
```
